tests/integration/test_guard_configurations.py

```diff
@@ -0,0 +1,322 @@
+        yield custom_model
+
+
+@pytest.fixture
+def guard_deployment_toxicity(
+    dr_endpoint: str, dr_token: str, default_prediction_server_id: str, cleanup_dr: Any
+) -> Generator[str, None, None]:
+    registered_model_name = "Toxicity Classifier"
+    deployment_name = "Toxicity Deployment pytest"
+
+    with cleanup_dr("deployments"):
+        for version in dr.RegisteredModel.list(search=registered_model_name)[0].list_versions():  # type: ignore
+            if version.build_status == "failed":
+                continue
+            registered_model_vers_id = version.id
+            break
+        deployment = get_or_create_deployment_from_registered_model_version(
+            endpoint=dr_endpoint,
+            token=dr_token,
+            registered_model_version_id=registered_model_vers_id,
+            label=deployment_name,
+            default_prediction_server_id=default_prediction_server_id,
+        )
+        yield deployment
+
+
+@pytest.fixture
+def guard_template_name_toxicity() -> str:
+    return "Toxicity"
+
+
+@pytest.fixture
+def intervention() -> dict[str, Any]:
+    return {
+        "action": "block",
+        "message": "This is a block message.",
+        "conditions": [{"comparand": 0.5, "comparator": "greaterThan"}],
+        "send_notification": True,
+    }
+
+
+@pytest.fixture
+def intervention_2() -> dict[str, Any]:
+    return {
+        "action": "block",
+        "message": "This is a different block message.",
+        "conditions": [{"comparand": 0.5, "comparator": "greaterThan"}],
+        "send_notification": True,
+    }
+
+
+@pytest.fixture
+def guard_deployment_prompt_injection(
+    dr_endpoint: str, dr_token: str, default_prediction_server_id: str, cleanup_dr: Any
+) -> Generator[str, None, None]:
+    registered_model_name = "Prompt Injection Classifier"
+    deployment_name = "Prompt Injection Deployment pytest"
+
+    with cleanup_dr("deployments"):
+        for version in dr.RegisteredModel.list(search=registered_model_name)[0].list_versions():  # type: ignore
+            if version.build_status == "failed":
+                continue
+            registered_model_vers_id = version.id
+            break
+        deployment = get_or_create_deployment_from_registered_model_version(
+            endpoint=dr_endpoint,
+            token=dr_token,
+            registered_model_version_id=registered_model_vers_id,
+            label=deployment_name,
+            default_prediction_server_id=default_prediction_server_id,
+        )
+        yield deployment
+
+
+@pytest.fixture
+def guard_template_name_prompt_injection() -> str:
+    return "Prompt Injection"
+
+
+def _get_guard_configs(custom_model_version_id: str) -> Any:
+    client = dr.client.get_client()
+    return client.get(
+        "guardConfigurations/",
+        params={"entityId": custom_model_version_id, "entityType": "customModelVersion"},
+    ).json()["data"]
+
+
+def test_get_or_create_guard_config_to_custom_model_version(
```

Comment:
minor: suggest renaming the two test functions to better align to the new helper names